---
title: "ITS2_analysis"
author: "ross"
date: "5/1/2023"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

```{r}
library(permute)
library(RColorBrewer)
library(janitor)
library(phyloseq)
library(tidyverse)
library(ggh4x)
library(cowplot)
library(naniar)
library(vegan)
library(emmeans)
```


# Load Data

### Load SymPortal ITS2 profiles

```{r profiles}
# Load ITS2 type profiles

# Tax table
tax0 <- read_tsv(
  file = "../ITS2_sequencing/symportal_output/its2_type_profiles/260_20230428T204810_DBV_20230429T054819.profiles.absolute.abund_and_meta.txt",
  n_max = 6) %>%
  dplyr::select(-2) %>% 
  gather(UID, value, -1) %>% 
  spread(1, value) %>%
  clean_names()
tax1 <- as.matrix(tax0[, -1], dimnames = list(tax0$uid, colnames(tax0[-1])))
rownames(tax1) <- tax0$uid
tax <- tax_table(tax1)

# OTU table
otu0 <- read_tsv(
  file = "../ITS2_sequencing/symportal_output/its2_type_profiles/260_20230428T204810_DBV_20230429T054819.profiles.absolute.abund_and_meta.txt") %>% 
  rename(sample_name = `...2`) %>%
  #rename(sample_name = X2) %>% #you may need to run this version of rename depending on how the file is loaded and column names are assigned
  select(-1) %>%
  slice(7:n()) %>%
  mutate_at(2:ncol(.), as.numeric)
otu1 <- as.matrix(otu0[, -1])
rownames(otu1) <- otu0$sample_name
otu <- otu_table(otu1, taxa_are_rows = FALSE)

# Sample data
sam0 <- read_csv("../ITS2_sequencing/20230425_Cunning_Symportal_datasheet.csv") %>%
  clean_names() %>%
  mutate(colony_id = str_extract(sample_name, pattern = "..._[^_]*"),
         timepoint = str_extract(sample_name, pattern = "[^_]*$"))

# colonies without site/species metadata were typos or incorrect labeling. AH has manually changed them here 
# AH has reconciled 6 of the 11 missing colony ID's

sam0 <- sam0 %>%
  mutate(colony_id=if_else(colony_id=="POR_44", "POC_44", 
                           if_else(colony_id=="POR_369", "POC_369", 
                           if_else(colony_id=="POC_83", "POR_83", 
                           if_else(colony_id=="POC_240", "POR_240", 
                           if_else(colony_id=="ACR_398", "ACR_396", 
                           if_else(colony_id=="ACR_314", "ACR_374", colony_id)))))))

# LEFT TO IDENTIFY
# POR-352
# POC-295
# POC-25
# POC-234
# ACR-268


#load metadata 
md <- read_csv("../metadata/coral_metadata.csv") %>%
  mutate(colony_id = str_replace(colony_id, "-", "_"))

sam0 <- sam0 %>%
  left_join(md, by = "colony_id")

#show colonies without species and site metadata 
na_colonies<-sam0%>%
            filter(is.na(species))

sam1 <- as.matrix(sam0)

rownames(sam1) <- sam0$sample_name

sam <- sample_data(data.frame(sam1))

profiles <- phyloseq(otu, tax, sam)
```


### Load SymPortal post-MED sequences

```{r variants}
# Load all post-MED sequence variants

# Tax table
taxnames <- read_tsv(
  file = "../ITS2_sequencing/symportal_output/post_med_seqs/260_20230428T204810_DBV_20230429T054819.seqs.absolute.abund_and_meta.txt",
  n_max = 0) %>%
  select(-(1:39)) %>%
  names(.)
tax0 <- tibble(
  DIV = taxnames,
  clade = str_extract(DIV, "[A-Z]")
)
tax1 <- as.matrix(tax0)
rownames(tax1) <- tax0$DIV
tax <- tax_table(tax1)

# OTU table
otu0 <- read_tsv(
  file = "../ITS2_sequencing/symportal_output/post_med_seqs/260_20230428T204810_DBV_20230429T054819.seqs.absolute.abund_and_meta.txt") %>% 
  select(-1, -(3:39))
otu1 <- as.matrix(otu0[, -1])
rownames(otu1) <- otu0$sample_name
otu <- otu_table(otu1, taxa_are_rows = FALSE)

variants <- phyloseq(otu, tax, sam)
```

### Subset just E5 timeseries samples
```{r filter}
# Subset to just get the E5 samples (separate others that were included in the same sequencing run)
e5v <- subset_samples(variants, grepl("^POC|^POR|^ACR", sample_name))
e5v <- prune_taxa(taxa_sums(e5v) != 0, e5v)

e5p <- subset_samples(profiles, grepl("^POC|^POR|^ACR", sample_name))
e5p <- prune_taxa(taxa_sums(e5p) != 0, e5p)
```

# Sequencing summary

```{r}
# Total number of reads
sum(sample_sums(e5v))

# Number of reads per sample (SymPortal post-MED sequences)
hist(log10(sample_sums(e5v)))

# Number of samples with >5000 reads
table(sample_sums(e5v) > 5000)

# Number of samples with >10000 reads
table(sample_sums(e5v) > 10000)

# Which samples had lowest read counts?
sample_sums(e5v)[order(sample_sums(e5v))][1:10]

# Number of unique sequence variants
ntaxa(e5v)

# Number of ITS2 type profiles
ntaxa(e5p)
```

# Results: plot sequences and profiles for all colonies over time

```{r}
## Transform variant and profile counts to relative abundance
v <- transform_sample_counts(e5v, function(x) x/sum(x))     # Relative abundance
p <- transform_sample_counts(e5p, function(x) x/sum(x))     # Relative abundance

## Filter out variants less than 0.1% relative abundance for plotting
vt <- transform_sample_counts(v, function(x) ifelse(x > 0.001, x, 0))
vt <- prune_taxa(taxa_sums(vt) != 0, vt)


## Set colors for plotting variants, with distinct palettes by clade/genus
tt <- as_tibble(data.frame(tax_table(vt))) %>%
  arrange(clade, DIV)


#change colors by type profile (e.g., anything with C1 at the beginning is one color, anything with C3 at the beginning is another, etc)
nt <- tt %>% count(clade) %>% pivot_wider(names_from = clade, values_from = n)
acols <- colorRampPalette(brewer.pal(9, "Greens"))(nt$A)
bcols <- colorRampPalette(brewer.pal(9, "Reds"))(nt$B)
ccols <- colorRampPalette(brewer.pal(9, "Blues"))(nt$C)
dcols <- colorRampPalette(brewer.pal(9, "Oranges"))(nt$D)
fcols <- colorRampPalette(brewer.pal(9, "Purples"))(nt$`F`)
gcols <- colorRampPalette(brewer.pal(9, "Greys"))(nt$G)
set.seed(559)
vpal <- c(acols[shuffle(acols)], bcols, ccols[shuffle(ccols)], 
          dcols[shuffle(dcols)], fcols[shuffle(fcols)], gcols[shuffle(gcols)])
vpal <- setNames(vpal, tt$DIV)

# Order variant data for plotting
divord <- as_tibble(data.frame(tax_table(vt))) %>%
  arrange(clade, DIV) %>%
  mutate(DIV = as_factor(DIV))

  
vmdf <- psmelt(vt) %>%
  filter(Abundance > 0) %>%
  arrange(host_genus, site, colony_id, timepoint, clade, DIV) %>%
  mutate(colony_id = as_factor(colony_id),
         sample_id = as_factor(paste(host_genus, colony_id, timepoint, sep = "_")),
         DIV = factor(DIV, levels = divord$DIV))


# Repeat for profiles
### truncate its2_type_profile names
pp <- as_tibble(data.frame(tax_table(p))) %>%
  arrange(clade, its2_type_profile) %>%
  mutate(its2_type_profile = str_trunc(its2_type_profile, 22, "right"))
np <- pp %>% count(clade) %>% pivot_wider(names_from = clade, values_from = n)
acols <- tail(brewer.pal(5, "Greens"), np$A)
#bcols <- colorRampPalette(brewer.pal(9, "YlGn"))(np$B)
ccols <- colorRampPalette(brewer.pal(9, "PuBu"))(np$C)
dcols <- tail(brewer.pal(9, "Oranges"), np$D)
gcols <- "gray40"
set.seed(559)
ppal <- c(acols, ccols, dcols, gcols)
ppal <- setNames(ppal, pp$its2_type_profile)

proford <- as_tibble(data.frame(tax_table(p))) %>%
  mutate(test = str_trunc(its2_type_profile, 46, "right"))%>%
  arrange(clade, its2_type_profile) %>%
  mutate(its2_type_profile = as_factor(its2_type_profile))
  
pmdf <- psmelt(p) %>% 
  
  arrange(host_genus, site, colony_id, timepoint, clade, its2_type_profile) %>%
  mutate(host_id = as_factor(colony_id),
         its2_type_profile = str_trunc(its2_type_profile, 46, "right"),
         sample_id = as_factor(paste(host_genus, colony_id, timepoint, sep = "_"))) %>% 
  filter(Abundance > 0.01)

# Plot variants
varfig <- vmdf %>%
  ggplot(aes(x = timepoint, y = Abundance, fill = DIV, label = DIV)) + 
  geom_col(width = 1) + 
  facet_nested(~ site + colony_id,
               scales = "free_x", space = "free_x",
               nest_line = element_line(linetype = 1)) +
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0, 0), labels = scales::percent) +
  ylab("ITS2 sequences") +
  scale_fill_manual(values = vpal[vmdf$OTU]) +
  theme(legend.position = "none", panel.background = element_blank(), 
        text = element_text(size = 9), plot.margin = margin(5.5, 5.5, 1, 5.5),
        strip.text.x = element_text(margin = margin(0.5, 0, 0.5, 0, "mm"), face = "italic"),
        axis.title.x = element_blank(), axis.text.y = element_blank(),
        axis.ticks = element_blank(),
        panel.spacing = unit(0.25, "mm"),
        strip.background = element_blank())

# Plot profiles
profig <- pmdf %>% 
  ggplot(aes(x = timepoint, y = Abundance)) +
    geom_col(width = 1, aes(fill = its2_type_profile)) +
    facet_nested(~ site + colony_id, scales = "free_x", space = "free_x") +
    scale_x_discrete(expand = c(0, 0)) +
    scale_y_continuous(expand = c(0, 0)) +
    ylab("Profiles") +
    scale_fill_manual(values = ppal[pmdf$its2_type_profile], limits = force, drop = TRUE) +
    theme(legend.position = "bottom", legend.key.size = unit(2, "mm"), legend.title = element_blank(),
          legend.margin = margin(1, 0, 0, 0), legend.box.margin = margin(-10, -10, 0, -10),
          legend.key = element_rect(fill = NA),
          plot.margin = margin(0, 5.5, 5.5, 5.5),
          text = element_text(size = 9), legend.text = element_text(size = 7),
          panel.background = element_blank(), strip.text.x = element_blank(),
          axis.title.x = element_blank(), axis.text.y = element_blank(),
          axis.ticks = element_blank(),
          strip.background = element_blank(),
          panel.spacing = unit(0.25, "mm")) +
    guides(fill = guide_legend(ncol = 8))
```

### Acropora
```{r, fig.width = 10, fig.height = 5}
# Site 1
acr_site1_var <- varfig %+% filter(vmdf, host_genus == "acropora", site == "site1")
acr_site1_pro <- profig %+% filter(pmdf, host_genus == "acropora", site == "site1")

acr_site1_plot <- plot_grid(acr_site1_var, acr_site1_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Site 2
acr_site2_var <- varfig %+% filter(vmdf, host_genus == "acropora", site == "site2")
acr_site2_pro <- profig %+% filter(pmdf, host_genus == "acropora", site == "site2")

acr_site2_plot <- plot_grid(acr_site2_var, acr_site2_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Site 3
acr_site3_var <- varfig %+% filter(vmdf, host_genus == "acropora", site == "site3")
acr_site3_pro <- profig %+% filter(pmdf, host_genus == "acropora", site == "site3")

acr_site3_plot <- plot_grid(acr_site3_var, acr_site3_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Acropora samples with no site info -- whatʻs up with these? typo in sample name?
as_tibble(sample_data(e5v)) %>%
  filter(host_genus == "acropora", is.na(site)) %>%
  pull(sample_name)

ggsave(file="Figures/ITS2/acr_site1.png", acr_site1_plot, width = 9, height = 7, units = c("in"))
ggsave(file="Figures/ITS2/acr_site2.png", acr_site2_plot, width = 9, height = 7, units = c("in"))
ggsave(file="Figures/ITS2/acr_site3.png", acr_site3_plot, width = 9, height = 7, units = c("in"))
```

### Porites
```{r, fig.width = 10, fig.height = 5}
# Site 1
por_site1_var <- varfig %+% filter(vmdf, host_genus == "porites", site == "site1")
por_site1_pro <- profig %+% filter(pmdf, host_genus == "porites", site == "site1")

por_site1_plot <- plot_grid(por_site1_var, por_site1_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Site 2
por_site2_var <- varfig %+% filter(vmdf, host_genus == "porites", site == "site2")
por_site2_pro <- profig %+% filter(pmdf, host_genus == "porites", site == "site2")

por_site2_plot <- plot_grid(por_site2_var, por_site2_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Site 3
por_site3_var <- varfig %+% filter(vmdf, host_genus == "porites", site == "site3")
por_site3_pro <- profig %+% filter(pmdf, host_genus == "porites", site == "site3")

por_site3_plot <- plot_grid(por_site3_var, por_site3_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Porites samples with no site info -- whatʻs up with these? typo in sample name?
as_tibble(sample_data(e5v)) %>%
  filter(host_genus == "porites", is.na(site)) %>%
  pull(sample_name)

ggsave(file="Figures/ITS2/por_site1.png", por_site1_plot, width = 9, height = 7, units = c("in"))
ggsave(file="Figures/ITS2/por_site2.png", por_site2_plot, width = 9, height = 7, units = c("in"))
ggsave(file="Figures/ITS2/por_site3.png", por_site3_plot, width = 9, height = 7, units = c("in"))
```


### Pocillopora
```{r, fig.width = 10, fig.height = 5}
# Site 1
poc_site1_var <- varfig %+% filter(vmdf, host_genus == "pocillopora", site == "site1")
poc_site1_pro <- profig %+% filter(pmdf, host_genus == "pocillopora", site == "site1")

poc_site1_plot <- plot_grid(poc_site1_var, poc_site1_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Site 2
poc_site2_var <- varfig %+% filter(vmdf, host_genus == "pocillopora", site == "site2")
poc_site2_pro <- profig %+% filter(pmdf, host_genus == "pocillopora", site == "site2")

poc_site2_plot <- plot_grid(poc_site2_var, poc_site2_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Site 3
poc_site3_var <- varfig %+% filter(vmdf, host_genus == "pocillopora", site == "site3")
poc_site3_pro <- profig %+% filter(pmdf, host_genus == "pocillopora", site == "site3")

poc_site3_plot <- plot_grid(poc_site3_var, poc_site3_pro, nrow = 2, label_fontface = "plain", rel_heights = c(0.6, 0.4))

# Pocillopora samples with no site info -- whatʻs up with these? typo in sample name?
as_tibble(sample_data(e5v)) %>%
  filter(host_genus == "pocillopora", is.na(site)) %>%
  pull(sample_name)

ggsave(file="Figures/ITS2/poc_site1.png", poc_site1_plot, width = 9, height = 7, units = c("in"))
ggsave(file="Figures/ITS2/poc_site2.png", poc_site2_plot, width = 9, height = 7, units = c("in"))
ggsave(file="Figures/ITS2/poc_site3.png", poc_site3_plot, width = 9, height = 7, units = c("in"))
```


# PCA and heatmap analyses 

## Creating heatmap 

For the time being we decided to use the color bar plots above. 

```{r}
pmdf_heatmap <- pmdf %>% 
  dplyr::select(colony_id, timepoint, site, species, its2_type_profile, Abundance) %>%
  ## need all samples to have values for each type profile
  spread(its2_type_profile, Abundance) %>%
  ## re:gather so that one column for type profile and one for abundance
  gather(its2_type_profile, Abundance, 5:44) #%>%
  ## changing NAs to zeros in Abundance column 
  #mutate(Abundance = coalesce(Abundance, 0))
  
pmdf_heatmap_plot <- pmdf_heatmap %>% subset(species == "Acropora") %>% ## CHANGE SPECIES HERE
  mutate(Abundance_count = coalesce(Abundance, 0)) %>%
  ## removing profiles that aren't present 
  group_by(species, its2_type_profile) %>%
  mutate(sum = sum(Abundance_count)) %>% ungroup() %>%
  filter(sum > 0) %>%
  replace_with_na_all(condition = ~.x == 0.000000000) %>%
  ggplot(., aes(x = timepoint, y = its2_type_profile)) + 
  geom_tile(aes(fill = Abundance), color = "black") +
  scale_fill_distiller(palette = "YlGn", na.value = "white", 
                       direction=100, labels = scales::label_percent(scale=100),
                       limits = c(min(pmdf_heatmap$Abundance), max(pmdf_heatmap$Abundance))) + 
  theme_classic() +
  theme(axis.text.x.bottom = element_text(angle = 60, hjust=1),
        axis.text.y = element_text(colour = 'black', size = 10, face = 'italic'),) + 
  #facet_grid(~site + timepoint, scales = "free") +
  facet_nested(~ site + colony_id, scales = "free_x", space = "free_x") +
  labs(x="Colony ID", fill="Relative Abundance", y="Clade")

#ggsave(file="Figures/ITS2/RelAb.png", pmdf_heatmap_plot, width = 14, height = 5, units = c("in"))
```


## PERMANOVA

Below is on relative abundance. Do VST later?

I'm removing the samples that we don't know a timepoint or site. 

```{r}
permanova_matrix <- pmdf_heatmap %>%
  mutate(Abundance = coalesce(Abundance, 0)) %>%
  spread(its2_type_profile, Abundance) %>%
  unite(ID, colony_id, timepoint, site, remove = FALSE) %>% na.omit()

permanova_meta <- permanova_matrix %>% 
  select(ID, colony_id, timepoint, site, species)

rownames(permanova_matrix) <- permanova_matrix[,1]
permanova_matrix <- permanova_matrix %>% select(-ID, -colony_id, -timepoint, -site, -species)

## below function takes awhile to run so output is copied below 
adonis2(permanova_matrix ~ species*timepoint*site, 
        data = permanova_meta, method='eu', na.rm = TRUE)

adonis2(permanova_matrix ~ species, 
        data = permanova_meta, method='eu', na.rm = TRUE)
```

Symbiont communities are different by species. 

Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = permanova_matrix ~ species, data = permanova_meta, method = "eu", na.rm = TRUE)
          Df SumOfSqs      R2      F Pr(>F)    
species    2   114.83 0.30905 98.625  0.001 ***
Residual 441   256.73 0.69095                  
Total    443   371.56 1.00000                  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Run for only PORITES
```{r}
permanova_matrix_por <- pmdf_heatmap %>%
  mutate(Abundance = coalesce(Abundance, 0)) %>%
  spread(its2_type_profile, Abundance) %>%
  unite(ID, colony_id, timepoint, site, remove = FALSE) %>% na.omit() %>%
  filter(species=="Porites")

permanova_meta_por <- permanova_matrix_por %>% 
  select(ID, colony_id, timepoint, site, species) %>%
  filter(species=="Porites")

rownames(permanova_matrix_por) <- permanova_matrix_por[,1]
permanova_matrix_por <- permanova_matrix_por %>% select(-ID, -colony_id, -timepoint, -site, -species)

#porites
  
adonis2(permanova_matrix_por ~ site*timepoint, 
        data = permanova_meta_por, method='eu', na.rm = TRUE)

```
Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = permanova_matrix_por ~ site * timepoint, data = permanova_meta_por, method = "eu", na.rm = TRUE)
                Df SumOfSqs      R2       F Pr(>F)    
site             2   14.146 0.12960 11.4609  0.001 ***
timepoint        3    0.592 0.00543  0.3200  0.997    
site:timepoint   6    1.222 0.01119  0.3299  0.999    
Residual       151   93.188 0.85378                   
Total          162  109.149 1.00000                   

Run for only POCILLOPORA
```{r}
permanova_matrix_poc <- pmdf_heatmap %>%
  mutate(Abundance = coalesce(Abundance, 0)) %>%
  spread(its2_type_profile, Abundance) %>%
  unite(ID, colony_id, timepoint, site, remove = FALSE) %>% na.omit() %>%
  filter(species=="Pocillopora")%>%
  filter(!timepoint=="TP2a")%>%
  filter(!timepoint=="TP2b")%>%
  droplevels()

permanova_meta_poc <- permanova_matrix_poc %>% 
  select(ID, colony_id, timepoint, site, species) %>%
  filter(species=="Pocillopora")%>%
  filter(!timepoint=="TP2a")%>%
  filter(!timepoint=="TP2b")%>%
  droplevels()

rownames(permanova_matrix_poc) <- permanova_matrix_poc[,1]
permanova_matrix_poc <- permanova_matrix_poc %>% select(-ID, -colony_id, -timepoint, -site, -species)

#porites
  
adonis2(permanova_matrix_poc ~ site*timepoint, 
        data = permanova_meta_poc, method='eu', na.rm = TRUE)

```
                Df SumOfSqs      R2      F Pr(>F)    
site             2    5.691 0.04633 3.6631  0.001 ***
timepoint        3    0.999 0.00813 0.4286  0.967    
site:timepoint   6    2.747 0.02236 0.5893  0.952    
Residual       146  113.408 0.92319                  
Total          157  122.845 1.00000                     

Run for only ACROPORA
```{r}
permanova_matrix_acr <- pmdf_heatmap %>%
  mutate(Abundance = coalesce(Abundance, 0)) %>%
  spread(its2_type_profile, Abundance) %>%
  unite(ID, colony_id, timepoint, site, remove = FALSE) %>% na.omit() %>%
  filter(species=="Acropora")%>%
  filter(!timepoint=="TP2a")%>%
  filter(!timepoint=="TP2b")%>%
  droplevels()

permanova_meta_acr <- permanova_matrix_acr %>% 
  select(ID, colony_id, timepoint, site, species) %>%
  filter(species=="Acropora")%>%
  filter(!timepoint=="TP2a")%>%
  filter(!timepoint=="TP2b")%>%
  droplevels()

rownames(permanova_matrix_acr) <- permanova_matrix_acr[,1]
permanova_matrix_acr <- permanova_matrix_acr %>% select(-ID, -colony_id, -timepoint, -site, -species)
  
adonis2(permanova_matrix_acr ~ site*timepoint, 
        data = permanova_meta_acr, method='eu', na.rm = TRUE)

```
                Df SumOfSqs      R2      F Pr(>F)
site             2   0.1860 0.00791 0.4396  0.765
timepoint        3   0.2992 0.01273 0.4715  0.828
site:timepoint   6   0.3832 0.01630 0.3018  0.986
Residual       107  22.6385 0.96306              
Total          118  23.5069 1.00000     


## NMDS 

Reference: https://jkzorz.github.io/2019/06/06/NMDS.html 

```{r}
permanova_matrix_com = as.matrix(permanova_matrix)
nmds = metaMDS(permanova_matrix_com, distance = "bray")

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
data.scores = as.data.frame(scores(nmds)$sites)

#add columns to data frame 
data.scores$colony_id = permanova_meta$colony_id
data.scores$timepoint = permanova_meta$timepoint
data.scores$site = permanova_meta$site
data.scores$species = permanova_meta$species

data.scores<-data.scores%>%
  filter(!timepoint=="TP2a")%>%
  filter(!timepoint=="TP2b")
 
head(data.scores)
set.seed(15647)
NMDS_plot_site <- ggplot(data.scores, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 2, aes(fill = site), pch=21, color="black")+ 
    #stat_ellipse(aes(colour=site), level=0.95, type="norm")+
    theme(axis.text.y = element_text(colour = "black", size = 10, face = "bold"), 
          axis.text.x = element_text(colour = "black", face = "bold", size = 10), 
          legend.text = element_text(size = 10, face ="bold", colour ="black"), 
          legend.position = "right", axis.title.y = element_text(face = "bold", size = 12), 
          axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
          legend.title = element_text(size = 12, colour = "black", face = "bold"), 
          panel.background = element_blank(), 
          panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
          legend.key=element_blank()) + 
          labs(x = "NMDS1", fill = "Site", y = "NMDS2")  + 
    scale_fill_manual(values=c("#ff6633", "#374d7c", "#00cccc"), labels=c("Manava", "Mahana", "Hilton")) +
  facet_wrap(~species, scales = "free");NMDS_plot_site

ggsave(file="Figures/ITS2/NMDS_site.png", NMDS_plot_site, width = 10, height = 5, units = c("in"))
```

Stress output from NMDS: 

Run 0 stress 0.01154461 
Run 1 stress 0.006580659 
... New best solution
... Procrustes: rmse 0.04461615  max resid 0.1714107 
Run 2 stress 0.005974172 
... New best solution
... Procrustes: rmse 0.04556396  max resid 0.4160293 
Run 3 stress 0.006684201 
Run 4 stress 0.005883244 
... New best solution
... Procrustes: rmse 0.0450752  max resid 0.2643378 
Run 5 stress 0.006431599 
Run 6 stress 0.007868632 
Run 7 stress 0.00574545 
... New best solution
... Procrustes: rmse 0.04552407  max resid 0.3412685 
Run 8 stress 0.005285655 
... New best solution
... Procrustes: rmse 0.0458617  max resid 0.3027762 
Run 9 stress 0.007687034 
Run 10 stress 0.005847585 
Run 11 stress 0.005768261 
... Procrustes: rmse 0.04077552  max resid 0.3717517 
Run 12 stress 0.005215554 
... New best solution
... Procrustes: rmse 0.04484562  max resid 0.3062907 
Run 13 stress 0.00787308 
Run 14 stress 0.006540272 
Run 15 stress 0.008656133 
Run 16 stress 0.01030515 
Run 17 stress 0.00737306 
Run 18 stress 0.007684081 
Run 19 stress 0.005174734 
... New best solution
... Procrustes: rmse 0.04394551  max resid 0.3224025 
Run 20 stress 0.007133301 
*** Best solution was not repeated -- monoMDS stopping criteria:
    20: no. of iterations >= maxit

output: 

all:
metaMDS(comm = permanova_matrix_com, distance = "bray") 

global Multidimensional Scaling using monoMDS

Data:     permanova_matrix_com 
Distance: bray 

Dimensions: 2 
Stress:     0.005174734 
Stress type 1, weak ties
Best solution was not repeated after 20 tries
The best solution was from try 19 (random start)
Scaling: centring, PC rotation, halfchange scaling 
Species: expanded scores based on ‘permanova_matrix_com’

**For a good representation of your data, the stress value should ideally be less than 0.2** 

Plot by timepoint
```{r}
permanova_matrix_com = as.matrix(permanova_matrix)
nmds = metaMDS(permanova_matrix_com, distance = "bray")

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
data.scores = as.data.frame(scores(nmds)$site)

#add columns to data frame 
data.scores$colony_id = permanova_meta$colony_id
data.scores$timepoint = permanova_meta$timepoint
data.scores$site = permanova_meta$site
data.scores$species = permanova_meta$species

data.scores<-data.scores%>%
  filter(!timepoint=="TP2a")%>%
  filter(!timepoint=="TP2b")
 
head(data.scores)
set.seed(15647)
NMDS_plot_time <- ggplot(data.scores, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 2, aes(fill = timepoint), color="black", pch=21)+ 
    #stat_ellipse(aes(colour=site), level=0.95, type="norm")+
    theme(axis.text.y = element_text(colour = "black", size = 10, face = "bold"), 
          axis.text.x = element_text(colour = "black", face = "bold", size = 10), 
          legend.text = element_text(size = 10, face ="bold", colour ="black"), 
          legend.position = "right", axis.title.y = element_text(face = "bold", size = 12), 
          axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
          legend.title = element_text(size = 12, colour = "black", face = "bold"), 
          panel.background = element_blank(), 
          panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
          legend.key=element_blank()) + 
          labs(x = "NMDS1", fill = "Timepoint", y = "NMDS2")  + 
        scale_fill_manual(values=c("white", "lightgray", "darkgray", "black"), labels=c("Jan 2020", "Mar 2020", "Sep 2020", "Nov 2020")) +
  facet_wrap(~species, scales = "free");NMDS_plot_time

ggsave(file="Figures/ITS2/NMDS_time.png", NMDS_plot_time, width = 10, height = 5, units = c("in"))
```

Plot by species
```{r}
permanova_matrix_com = as.matrix(permanova_matrix)
nmds = metaMDS(permanova_matrix_com, distance = "bray")

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
data.scores = as.data.frame(scores(nmds)$site)

#add columns to data frame 
data.scores$colony_id = permanova_meta$colony_id
data.scores$timepoint = permanova_meta$timepoint
data.scores$site = permanova_meta$site
data.scores$species = permanova_meta$species

data.scores<-data.scores%>%
  filter(!timepoint=="TP2a")%>%
  filter(!timepoint=="TP2b")
 
head(data.scores)
set.seed(15647)
NMDS_plot_species <- ggplot(data.scores, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 2, aes(fill = species), color="black", pch=21)+ 
    #stat_ellipse(aes(colour=site), level=0.95, type="norm")+
    theme(axis.text.y = element_text(colour = "black", size = 10, face = "bold"), 
          axis.text.x = element_text(colour = "black", face = "bold", size = 10), 
          legend.text = element_text(size = 10, face ="bold", colour ="black"), 
          legend.position = "right", axis.title.y = element_text(face = "bold", size = 12), 
          axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
          legend.title = element_text(size = 12, colour = "black", face = "bold"), 
          panel.background = element_blank(), 
          panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
          legend.key=element_blank()) + 
          labs(x = "NMDS1", fill = "Species", y = "NMDS2")  + 
          scale_fill_manual(values=c("gray", "orange", "purple")); NMDS_plot_species

ggsave(file="Figures/ITS2/NMDS_species.png", NMDS_plot_species, width = 5, height = 4, units = c("in"))
```

Output matrix for additional analyses. 
```{r}
output_matrix <- pmdf_heatmap %>%
  mutate(Abundance = coalesce(Abundance, 0)) %>%
  spread(its2_type_profile, Abundance)%>%
  unite(ID, colony_id, timepoint, site, remove = FALSE) %>% na.omit()%>%
  mutate(colony_id = str_replace_all(colony_id, "_", "-"))%>%
  mutate(colony_id_corr = colony_id)%>%
  filter(!timepoint=="TP2b")%>%
  mutate(timepoint=if_else(timepoint=="TP1", "timepoint1", 
                           if_else(timepoint=="TP2", "timepoint2", 
                                   if_else(timepoint=="TP3", "timepoint3", 
                                           if_else(timepoint=="TP4", "timepoint4", 
                                                   if_else(timepoint=="TP2a", "timepoint2", 
                                                           if_else(timepoint=="TP2b", "timepoint2", NA)))))))%>%
  mutate(site=if_else(site=="site1", "Manava", 
                      if_else(site=="site2", "Mahana", 
                              if_else(site=="site3", "Hilton", NA))))%>%
  mutate(nutrient=if_else(site=="Manava", "High", 
                      if_else(site=="Mahana", "Low", 
                              if_else(site=="Hilton", "Medium", NA))))%>%
  mutate(site_code=if_else(site=="Manava", "Manava High", 
                      if_else(site=="Mahana", "Mahana Low", 
                              if_else(site=="Hilton", "Hilton Medium", NA))))%>%
  mutate(month=if_else(timepoint=="timepoint1", "January 2020", 
                       if_else(timepoint=="timepoint2", "March 2020", 
                               if_else(timepoint=="timepoint3", "September 2020", 
                                       if_else(timepoint=="timepoint4", "November 2020", NA)))))%>%
  select(!ID)%>%
  select(colony_id, colony_id_corr, timepoint, site, nutrient, site_code, month, everything())
  
output_matrix<-as.data.frame(output_matrix)

write_csv(output_matrix, file="Output/ITS2_rel_abund_matrix.csv")
```

Output matrix for E5 Deep Dive samples. 
```{r}
head(output_matrix) 

deep_dive_colonies<-c("ACR-140", "ACR-145", "ACR-150", "ACR-173", "ACR-178", "POC-47", "POC-48", "POC-50", "POC-53", "POC-57", "POR-71", "POR-73", "POR-76", "POR-79", "POR-82")

deep_dive<-output_matrix%>%
  filter(colony_id_corr %in% deep_dive_colonies)%>%
  arrange(colony_id_corr)%>%
  filter(timepoint=="timepoint2")%>%
  write.csv(., "Output/deep_dive_ITS2_matrix.csv")
```


# Plot ITS2 averaged for each species 
```{r}
#create custom color palette 
get_custom_color <- function(Profile) {
  if (startsWith(Profile, "A1")) {
    return("brown4")
  } else if (startsWith(Profile, "C1/")) {
    return("lightblue")
  } else if (startsWith(Profile, "C116")) {
    return("lightblue3")
  #} else if (startsWith(Profile, "C15")) {
  #  return("cyan4")
  } else if (startsWith(Profile, "C15-C15")) {
    return("cyan4")
  } else if (startsWith(Profile, "C15/C15")) {
    return("cyan4")
  } else if (startsWith(Profile, "C15/C116")) {
    return("cyan3")
  } else if (startsWith(Profile, "C15/C93")) {
    return("cyan2")
  } else if (startsWith(Profile, "C15h/C3")) {
    return("cyan1")
  } else if (startsWith(Profile, "C1d")) {
    return("lightgray")
  } else if (startsWith(Profile, "C3")) {
    return("darkgray")
  } else if (startsWith(Profile, "C42")) {
    return("lightblue4")
  } else if (startsWith(Profile, "C91")) {
    return("blue4")
  } else if (Profile=="D1") {
    return("orange")
  } else if (startsWith(Profile, "D1-D17")) {
    return("orange")
  } else if (startsWith(Profile, "D1-D1")) {
    return("orange")
  } else if (startsWith(Profile, "D1/D4")) {
    return("orange3")
  } else if (startsWith(Profile, "G3")) {
    return("white")
  } else {
    return("black")  # Return a default color for other cases, if desired
  }
}

output_matrix$site<-factor(output_matrix$site, levels=c("Mahana", "Hilton", "Manava"))

plot_species_df<-output_matrix%>%
  group_by(species)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  pivot_longer(-species, values_to="Abundance", names_to="Profile")
  
plot_species<-ggplot(plot_species_df, aes(x = species, y = Abundance)) + 
  geom_bar(position="stack", stat="identity", aes(fill = Profile), color = "black") +
  scale_fill_manual(values = sapply(plot_species_df$Profile, get_custom_color))+
  theme_classic() +
  theme(axis.text.y = element_text(colour = 'black', size = 10), 
        axis.title = element_text(colour = 'black', size = 12, face="bold"), 
        axis.text.x = element_text(colour = 'black', size = 10, angle = 60, hjust=1, face="italic")
        ) + 
  labs(x="Species", fill="ITS2 Profile", y="Relative Abundance");plot_species
  
#ggsave(plot_species, filename="Figures/ITS2/species_profiles.png", height=6, width=14)
#ggsave(plot_species, filename="Figures/ITS2/species_profiles.jpeg", height=6, width=14)
#ggsave(plot_species, filename="Figures/ITS2/species_profiles.pdf", height=6, width=14)
```

# Plot ITS2 averaged for each species x site 
```{r}
plot_species_site_df<-output_matrix%>%
  group_by(species, site)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  pivot_longer(-c(species,site), values_to="Abundance", names_to="Profile")
  
plot_species_site<-ggplot(plot_species_site_df, aes(x = site, y = Abundance)) + 
  facet_wrap(~species)+
  geom_bar(position="stack", stat="identity", aes(fill = Profile), color = "black") +
  scale_fill_manual(values = sapply(plot_species_df$Profile, get_custom_color))+
  theme_classic() +
  theme(axis.text.y = element_text(colour = 'black', size = 10), 
        axis.title = element_text(colour = 'black', size = 12, face="bold"), 
        axis.text.x = element_text(colour = 'black', size = 10, angle = 60, hjust=1),
        strip.text =element_text(colour="black", face="italic", size=8), 
        ) + 
  
  #facet_grid(~site + timepoint, scales = "free") +
  #facet_nested(~ species, scales = "free_x", space = "free_x") +
  labs(x="Site", fill="ITS2 Profile", y="Relative Abundance");plot_species_site

#ggsave(plot_species_site, filename="Figures/ITS2/species_site_profiles.png", height=6, width=14)
#ggsave(plot_species_site, filename="Figures/ITS2/species_site_profiles.jpeg", height=6, width=14)
#ggsave(plot_species_site, filename="Figures/ITS2/species_site_profiles.pdf", height=6, width=14)
```

# Plot individual species separately with all ITS2 profiles in unique colors 

Acropora 
```{r}
#create custom color palette 
get_custom_color_acr <- function(Profile) {
  if (Profile=="A1/A1ee") {
    return("darkgray")
  } else if (startsWith(Profile, "A1/A1ee-A1ep")) {
    return("lightgray")
  } else if (startsWith(Profile, "A1/A1gb-A1ee")) {
    return("black")
  } else if (startsWith(Profile, "D1")) {
    return("slategray")
  } else {
    return("black")  # Return a default color for other cases, if desired
  }
}

plot_species_site_df_acr<-output_matrix%>%
  filter(species=="Acropora")%>%
  group_by(species, site)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) %>% #remove taxa that are below 1% relative abundance
  droplevels()%>%
  pivot_longer(-c(species,site), values_to="Abundance", names_to="Profile")
  
plot_species_site_acr<-ggplot(plot_species_site_df_acr, aes(x = site, y = Abundance)) + 
  geom_bar(position="stack", stat="identity", aes(fill = Profile), color = "black") +
  scale_fill_manual(values = sapply(plot_species_df$Profile, get_custom_color_acr))+
  theme_classic() +
  theme(axis.text.y = element_text(colour = 'black', size = 10), 
        axis.title = element_text(colour = 'black', size = 12, face="bold"), 
        axis.text.x = element_text(colour = 'black', size = 10, angle = 60, hjust=1),
        strip.text =element_text(colour="black", face="italic", size=8), 
        ) + 
  ggtitle(expression(italic("Acropora")))+
  
  #facet_grid(~site + timepoint, scales = "free") +
  #facet_nested(~ species, scales = "free_x", space = "free_x") +
  labs(x="Site", fill="ITS2 Profile", y="Relative Abundance");plot_species_site_acr

#ggsave(plot_species_site_acr, filename="Figures/ITS2/ACR_site_profiles.png", height=6, width=8)
#ggsave(plot_species_site_acr, filename="Figures/ITS2/ACR_site_profiles.jpeg", height=6, width=8)
#ggsave(plot_species_site_acr, filename="Figures/ITS2/ACR_site_profiles.pdf", height=6, width=8)

#view averages 
plot_species_site_df_acr_summary<-output_matrix%>%
  filter(species=="Acropora")%>%
  group_by(species)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) %>% #remove taxa that are below 1% relative abundance
  droplevels()%>%
  pivot_longer(-c(species), values_to="Abundance", names_to="Profile");plot_species_site_df_acr_summary
```

Run model of relative abundance for Acropora. 
```{r}
acr_model<-output_matrix%>%
  filter(species=="Acropora")%>%
  group_by(colony_id, species, site)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) %>% #remove taxa that are below 1% relative abundance
  droplevels()%>%
  pivot_longer(-c(colony_id, species,site), values_to="Abundance", names_to="Profile")%>%
  
  aov(Abundance~site*Profile, data=.)

summary(acr_model)
qqPlot(residuals(acr_model))

emm<-emmeans(acr_model, ~ site | Profile)
pairs(emm)
```

Pocillopora 
```{r}
#create custom color palette 
get_custom_color_poc <- function(Profile) {
  if (startsWith(Profile, "C1/C42.2")) {
    return("orange4")
  } else if (startsWith(Profile, "C1d-C42.2")) {
    return("darkorange")
  } else if (startsWith(Profile, "C1d/C1")) {
    return("orange")
  } else if (startsWith(Profile, "C1d/C42.2")) {
    return("darkgoldenrod3")
  } else if (startsWith(Profile, "C1d/C42g")) {
    return("chocolate1")
  } else if (startsWith(Profile, "C42a")) {
    return("darkgoldenrod1")
  } else if (startsWith(Profile, "C42g")) {
    return("yellow2")
  } else {
    return("black")  # Return a default color for other cases, if desired
  }
}

plot_species_site_df_poc<-output_matrix%>%
  filter(species=="Pocillopora")%>%
  group_by(species, site)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) %>% #remove taxa that are below 1% relative abundance
  droplevels()%>%
  pivot_longer(-c(species,site), values_to="Abundance", names_to="Profile")
  
plot_species_site_poc<-ggplot(plot_species_site_df_poc, aes(x = site, y = Abundance)) + 
  geom_bar(position="stack", stat="identity", aes(fill = Profile), color = "black") +
  scale_fill_manual(values = sapply(plot_species_df$Profile, get_custom_color_poc))+
  theme_classic() +
  theme(axis.text.y = element_text(colour = 'black', size = 10), 
        axis.title = element_text(colour = 'black', size = 12, face="bold"), 
        axis.text.x = element_text(colour = 'black', size = 10, angle = 60, hjust=1),
        strip.text =element_text(colour="black", face="italic", size=8))+ 
  ggtitle(expression(italic("Pocillopora")))+
  
  #facet_grid(~site + timepoint, scales = "free") +
  #facet_nested(~ species, scales = "free_x", space = "free_x") +
  labs(x="Site", fill="ITS2 Profile", y="Relative Abundance");plot_species_site_poc

#ggsave(plot_species_site_poc, filename="Figures/ITS2/POC_site_profiles.png", height=6, width=8)
#ggsave(plot_species_site_poc, filename="Figures/ITS2/POC_site_profiles.jpeg", height=6, width=8)
#ggsave(plot_species_site_poc, filename="Figures/ITS2/POC_site_profiles.pdf", height=6, width=8)
```

Run model of relative abundance for Pocillopora 
```{r}
poc_model<-output_matrix%>%
  filter(species=="Pocillopora")%>%
  group_by(colony_id, species, site)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) %>% #remove taxa that are below 1% relative abundance
  droplevels()%>%
  pivot_longer(-c(colony_id, species,site), values_to="Abundance", names_to="Profile")%>%
  
  aov(Abundance~site*Profile, data=.)

summary(poc_model)
qqPlot(residuals(poc_model))

emm<-emmeans(poc_model, ~ site | Profile)
pairs(emm)
```

Porites 
```{r}
#create custom color palette 
get_custom_color_por <- function(Profile) {
  if (startsWith(Profile, "C116")) {
    return("lightblue")
  } else if (Profile=="C15-C15bn-C15by") {
    return("purple4")
  } else if (Profile=="C15-C15bn-C15by-C15ai") {
    return("deeppink")
  } else if (Profile=="C15-C15bn-C15dq") {
    return("darkslateblue")
  } else if (startsWith(Profile, "C15-C15by")) {
    return("pink")
  } else if (startsWith(Profile, "C15-C15he")) {
    return("pink3")
  } else if (Profile=="C15-C15kl") {
    return("lightgray")
  } else if (Profile=="C15-C15kl-C15he") {
    return("darkgray")
  } else if (Profile=="C15-C15kl-C15he-C15vz") {
    return("darkblue")
  } else if (Profile=="C15-C15kl-C15pm") {
    return("white")
  } else if (Profile=="C15-C15l-C15bm-C15n") {
    return("darkviolet")
  } else if (Profile=="C15-C15l-C15n") {
    return("gray1")
  } else if (Profile=="C15-C15l-C15n-C15bb-C15.8") {
    return("violet")   
  } else if (Profile=="C15-C15wa-C15kl-C15he-C15vz") {
    return("darkmagenta")   
  } else if (Profile=="C15/C116") {
    return("darkorchid1")    
  } else if (Profile=="C15/C116-C15h") {
    return("aliceblue")    
  } else if (Profile=="C15/C15ev") {
    return("deeppink3")   
  } else if (startsWith(Profile, "C1d/C1")) {
    return("darkred")
  } else {
    return("red")  # Return a default color for other cases, if desired
  }
}

plot_species_site_df_por<-output_matrix%>%
  filter(species=="Porites")%>%
  group_by(species, site)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) %>% #remove taxa that are below 1% relative abundance
  droplevels()%>%
  pivot_longer(-c(species,site), values_to="Abundance", names_to="Profile")
  
plot_species_site_por<-ggplot(plot_species_site_df_por, aes(x = site, y = Abundance)) + 
  geom_bar(position="stack", stat="identity", aes(fill = Profile), color = "black") +
  scale_fill_manual(values = sapply(plot_species_df$Profile, get_custom_color_por))+
  theme_classic() +
  theme(axis.text.y = element_text(colour = 'black', size = 10), 
        axis.title = element_text(colour = 'black', size = 12, face="bold"), 
        axis.text.x = element_text(colour = 'black', size = 10, angle = 60, hjust=1),
        strip.text =element_text(colour="black", face="italic", size=8))+ 
  ggtitle(expression(italic("Porites")))+
  
  #facet_grid(~site + timepoint, scales = "free") +
  #facet_nested(~ species, scales = "free_x", space = "free_x") +
  labs(x="Site", fill="ITS2 Profile", y="Relative Abundance");plot_species_site_por

#ggsave(plot_species_site_por, filename="Figures/ITS2/POR_site_profiles.png", height=6, width=8)
#ggsave(plot_species_site_por, filename="Figures/ITS2/POR_site_profiles.jpeg", height=6, width=8)
#ggsave(plot_species_site_por, filename="Figures/ITS2/POR_site_profiles.pdf", height=6, width=8)
```

Run model of relative abundance for Porites 
```{r}
por_model<-output_matrix%>%
  filter(species=="Porites")%>%
  group_by(colony_id, species, site)%>%
  summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) %>% #remove taxa that are below 1% relative abundance
  droplevels()%>%
  pivot_longer(-c(colony_id, species,site), values_to="Abundance", names_to="Profile")%>%
  
  aov(Abundance~site*Profile, data=.)

summary(por_model)
qqPlot(residuals(por_model))

emm<-emmeans(por_model, ~ site | Profile)
pairs(emm)
```

Assemble plots. 
```{r}
its2_plots<-plot_grid(plot_species_site_acr, plot_species_site_poc, plot_species_site_por, ncol=1, nrow=3, align="vh")

ggsave(its2_plots, filename="Figures/ITS2/species_site_panel.jpeg", width=8, height=20)
ggsave(its2_plots, filename="Figures/ITS2/species_site_panel.pdf", width=8, height=20)
```

# Assemble NMDS plot for each species by site. 

Acropora
```{r}
nmds_df_acr<-output_matrix%>%
  filter(species=="Acropora")%>%
  group_by(colony_id, species, site)%>%
    summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) #remove taxa that are below 1% relative abundance

nmds_meta_acr <- nmds_df_acr %>% 
  select(colony_id, species, site)

nmds_df_acr<-nmds_df_acr%>%
  ungroup()%>%
  select(!colony_id)%>%
  select(!site)%>%
  select(!species)

nmds_df_acr = as.matrix(nmds_df_acr)
nmds_acr = metaMDS(nmds_df_acr, distance = "bray")

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
data.scores = as.data.frame(scores(nmds_acr)$sites)

#add columns to data frame 
data.scores$site = nmds_meta_acr$site
data.scores$species = nmds_meta_acr$species
 
head(data.scores)
set.seed(15647)
NMDS_plot_acr <- ggplot(data.scores, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 2, aes(colour = site))+ 
    #stat_ellipse(aes(colour=site), level=0.95, type="norm")+
    ggtitle(expression(italic("Acropora")))+
    theme_classic()+
    theme(axis.text.y = element_text(colour = "black", size = 10, face = "bold"), 
          axis.text.x = element_text(colour = "black", face = "bold", size = 10), 
          legend.text = element_text(size = 10, face ="bold", colour ="black"), 
          legend.position = "right", axis.title.y = element_text(face = "bold", size = 12), 
          axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
          legend.title = element_text(size = 12, colour = "black", face = "bold"), 
          panel.background = element_blank(), 
          panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
          legend.key=element_blank()) + 
          labs(x = "NMDS1", colour = "Site", y = "NMDS2")  + 
    scale_colour_manual(values=c("#ff6633", "#374d7c", "#00cccc"));NMDS_plot_acr

ggsave(file="Figures/ITS2/NMDS_Acr.png", NMDS_plot_acr, width = 5, height = 5, units = c("in"))
```

Pocillopora
```{r}
nmds_df_poc<-output_matrix%>%
  filter(species=="Pocillopora")%>%
  group_by(colony_id, species, site)%>%
    summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) #remove taxa that are below 1% relative abundance

nmds_meta_poc <- nmds_df_poc %>% 
  select(colony_id, species, site)

nmds_df_poc<-nmds_df_poc%>%
  ungroup()%>%
  select(!colony_id)%>%
  select(!site)%>%
  select(!species)

nmds_df_poc = as.matrix(nmds_df_poc)
nmds_poc = metaMDS(nmds_df_poc, distance = "bray")

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
data.scores = as.data.frame(scores(nmds_poc)$sites)

#add columns to data frame 
data.scores$site = nmds_meta_poc$site
data.scores$species = nmds_meta_poc$species
 
head(data.scores)
set.seed(15647)
NMDS_plot_poc <- ggplot(data.scores, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 2, aes(colour = site))+ 
    #stat_ellipse(aes(colour=site), level=0.95, type="norm")+
    ggtitle(expression(italic("Pocillopora")))+
    theme_classic()+
    theme(axis.text.y = element_text(colour = "black", size = 10, face = "bold"), 
          axis.text.x = element_text(colour = "black", face = "bold", size = 10), 
          legend.text = element_text(size = 10, face ="bold", colour ="black"), 
          legend.position = "right", axis.title.y = element_text(face = "bold", size = 12), 
          axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
          legend.title = element_text(size = 12, colour = "black", face = "bold"), 
          panel.background = element_blank(), 
          panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
          legend.key=element_blank()) + 
          labs(x = "NMDS1", colour = "Site", y = "NMDS2")  + 
    scale_colour_manual(values=c("#ff6633", "#374d7c", "#00cccc"));NMDS_plot_poc

ggsave(file="Figures/ITS2/NMDS_Poc.png", NMDS_plot_poc, width = 5, height = 5, units = c("in"))
```

Pocillopora
```{r}
nmds_df_por<-output_matrix%>%
  filter(species=="Porites")%>%
  group_by(colony_id, species, site)%>%
    summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) #remove taxa that are below 1% relative abundance

nmds_meta_por <- nmds_df_por %>% 
  select(colony_id, species, site)

nmds_df_por<-nmds_df_por%>%
  ungroup()%>%
  select(!colony_id)%>%
  select(!site)%>%
  select(!species)

nmds_df_por = as.matrix(nmds_df_por)
nmds_por = metaMDS(nmds_df_por, distance = "bray")

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
data.scores = as.data.frame(scores(nmds_por)$sites)

#add columns to data frame 
data.scores$site = nmds_meta_por$site
data.scores$species = nmds_meta_por$species
 
head(data.scores)
set.seed(15647)
NMDS_plot_por <- ggplot(data.scores, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 2, aes(colour = site))+ 
    #stat_ellipse(aes(colour=site), level=0.95, type="norm")+
    ggtitle(expression(italic("Porites")))+
    theme_classic()+
    theme(axis.text.y = element_text(colour = "black", size = 10, face = "bold"), 
          axis.text.x = element_text(colour = "black", face = "bold", size = 10), 
          legend.text = element_text(size = 10, face ="bold", colour ="black"), 
          legend.position = "right", axis.title.y = element_text(face = "bold", size = 12), 
          axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
          legend.title = element_text(size = 12, colour = "black", face = "bold"), 
          panel.background = element_blank(), 
          panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
          legend.key=element_blank()) + 
          labs(x = "NMDS1", colour = "Site", y = "NMDS2")  + 
    scale_colour_manual(values=c("#ff6633", "#374d7c", "#00cccc"));NMDS_plot_por

ggsave(file="Figures/ITS2/NMDS_Por.png", NMDS_plot_por, width = 5, height = 5, units = c("in"))
```

# Run PCA with Biplots for each species 

## Acropora
```{r}
pca_df_acr<-output_matrix%>%
  filter(species=="Acropora")%>%
  group_by(colony_id, species, site)%>%
    summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) #remove taxa that are below 1% relative abundance

scaled_acr<-prcomp(pca_df_acr[c(4:8)], scale=TRUE, center=TRUE) 
```

```{r}
# scale data
vegan_acr <- scale(pca_df_acr[ ,4:8])

# PerMANOVA - partitioning the euclidean distance matrix by species
permanova<-adonis2(vegan_acr ~ site, data = pca_df_acr, method='eu')
z_acr<-permanova
z_acr
``` 
 
```{r}
pca_acr<-ggplot2::autoplot(scaled_acr, data=pca_df_acr, frame.colour="site", loadings=FALSE,  colour="site", loadings.label.colour="black", loadings.colour="black", loadings.label=TRUE, frame=FALSE, loadings.label.size=5, loadings.label.vjust=-1, size=5, frame.type = 'norm', alpha=0.5) + 
  scale_colour_manual(values=c("#ff6633", "#374d7c", "#00cccc")) +
  scale_fill_manual(values=c("#ff6633", "#374d7c", "#00cccc")) + 
  theme_classic()+
  geom_text(x=0.25, y=-0.6, label=paste("p =", z_acr$`Pr(>F)`[1]), size=6, color=ifelse(z_acr$`Pr(>F)`[1] < 0.05, "black", "darkgray")) + 
  #xlim(-0.3,0.5)+
  #ylim(-0.3, 0.5)+
  xlab("PC1")+
  ylab("PC2")+
   theme(legend.text = element_text(size=18, face="italic"), 
         legend.position="right",
        plot.background = element_blank(),
        legend.title = element_blank(), 
        axis.text = element_text(size=18, color="black"), 
        axis.title = element_text(size=18, color="black"));pca_acr

ggsave(filename="Figures/ITS2/ACR_ITS2_PCA.pdf", plot=pca_acr, dpi=300, width=7, height=5, units="in")
```

## Pocillopora 

```{r}
pca_df_poc<-output_matrix%>%
  filter(species=="Pocillopora")%>%
  group_by(colony_id, species, site)%>%
    summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) #remove taxa that are below 1% relative abundance

scaled_poc<-prcomp(pca_df_poc[c(4:17)], scale=TRUE, center=TRUE) 
```

```{r}
# scale data
vegan_poc <- scale(pca_df_poc[ ,4:17])

# PerMANOVA - partitioning the euclidean distance matrix by species
permanova<-adonis2(vegan_poc ~ site, data = pca_df_poc, method='eu')
z_poc<-permanova
z_poc
``` 
 
```{r}
pca_poc<-ggplot2::autoplot(scaled_poc, data=pca_df_poc, frame.colour="site", loadings=FALSE,  colour="site", loadings.label.colour="black", loadings.colour="black", loadings.label=FALSE, frame=TRUE, loadings.label.size=5, loadings.label.vjust=-1, size=5, frame.type = 'norm', alpha=0.5) + 
  scale_colour_manual(values=c("#ff6633", "#374d7c", "#00cccc")) +
  scale_fill_manual(values=c("#ff6633", "#374d7c", "#00cccc")) + 
  theme_classic()+
  geom_text(x=0.2, y=-0.4, label=paste("p =", z_poc$`Pr(>F)`[1]), size=6, color=ifelse(z_poc$`Pr(>F)`[1] < 0.05, "black", "darkgray")) + 
  #xlim(-0.3,0.5)+
  #ylim(-0.3, 0.5)+
  xlab("PC1")+
  ylab("PC2")+
   theme(legend.text = element_text(size=18, face="italic"), 
         legend.position="right",
        plot.background = element_blank(),
        legend.title = element_blank(), 
        axis.text = element_text(size=18, color="black"), 
        axis.title = element_text(size=18, color="black"));pca_poc

ggsave(filename="Figures/ITS2/POC_ITS2_PCA.pdf", plot=pca_poc, dpi=300, width=7, height=5, units="in")
```

## Porites 

```{r}
pca_df_por<-output_matrix%>%
  filter(species=="Porites")%>%
  group_by(colony_id, species, site)%>%
    summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) #remove taxa that are below 1% relative abundance

scaled_por<-prcomp(pca_df_por[c(4:25)], scale=TRUE, center=TRUE) 
```

```{r}
# scale data
vegan_por <- scale(pca_df_por[ ,4:25])

# PerMANOVA - partitioning the euclidean distance matrix by species
permanova<-adonis2(vegan_por ~ site, data = pca_df_por, method='eu')
z_por<-permanova
z_por
``` 
 
```{r}
pca_por<-ggplot2::autoplot(scaled_por, data=pca_df_por, frame.colour="site", loadings=FALSE,  colour="site", loadings.label.colour="black", loadings.colour="black", loadings.label=FALSE, frame=FALSE, loadings.label.size=5, loadings.label.vjust=-1, size=5, frame.type = 'norm', alpha=0.5) + 
  scale_colour_manual(values=c("#ff6633", "#374d7c", "#00cccc")) +
  scale_fill_manual(values=c("#ff6633", "#374d7c", "#00cccc")) + 
  theme_classic()+
  geom_text(x=0.2, y=-0.4, label=paste("p =", z_por$`Pr(>F)`[1]), size=6, color=ifelse(z_por$`Pr(>F)`[1] < 0.05, "black", "darkgray")) + 
  #xlim(-0.3,0.5)+
  #ylim(-0.3, 0.5)+
  xlab("PC1")+
  ylab("PC2")+
   theme(legend.text = element_text(size=18, face="italic"), 
         legend.position="right",
        plot.background = element_blank(),
        legend.title = element_blank(), 
        axis.text = element_text(size=18, color="black"), 
        axis.title = element_text(size=18, color="black"));pca_por

ggsave(filename="Figures/ITS2/POR_ITS2_PCA.pdf", plot=pca_por, dpi=300, width=7, height=5, units="in")
```

## All species together 

```{r}
pca_df_all<-output_matrix%>%
  group_by(colony_id, species, site)%>%
    summarise_if(is.numeric, mean, na.rm = TRUE, .groups = "drop")%>%
  select_if(~ !is.numeric(.) || sum(.) >= 0.01) #remove taxa that are below 1% relative abundance

scaled_all<-prcomp(pca_df_all[c(4:40)], scale=TRUE, center=TRUE) 
```

```{r}
# scale data
vegan_all <- scale(pca_df_all[ ,4:40])

# PerMANOVA - partitioning the euclidean distance matrix by species
permanova<-adonis2(vegan_all ~ site * species, data = pca_df_all, method='eu')
z_all<-permanova
z_all
``` 
 
```{r}
pca_all<-ggplot2::autoplot(scaled_all, data=pca_df_all, frame.colour="species", loadings=FALSE,  colour="species", loadings.label.colour="black", loadings.colour="black", loadings.label=FALSE, frame=FALSE, loadings.label.size=5, loadings.label.vjust=-1, size=5, frame.type = 'norm', alpha=0.5) + 
  #facet_wrap(~species)+
  #scale_colour_manual(values=c("#ff6633", "#374d7c", "#00cccc")) +
  #scale_fill_manual(values=c("#ff6633", "#374d7c", "#00cccc")) + 
  theme_classic()+
  geom_text(x=0.25, y=-0.6, label=paste("site =", z_all$`Pr(>F)`[1]), size=6, color=ifelse(z_all$`Pr(>F)`[1] < 0.05, "black", "darkgray")) + 
    geom_text(x=0.25, y=-0.6, label=paste("species =", z_all$`Pr(>F)`[2]), size=6, color=ifelse(z_all$`Pr(>F)`[2] < 0.05, "black", "darkgray")) + 
    geom_text(x=0.25, y=-0.6, label=paste("interaction =", z_all$`Pr(>F)`[3]), size=6, color=ifelse(z_all$`Pr(>F)`[3] < 0.05, "black", "darkgray")) + 
  #xlim(-0.3,0.5)+
  #ylim(-0.3, 0.5)+
  xlab("PC1")+
  ylab("PC2")+
   theme(legend.text = element_text(size=18, face="italic"), 
         legend.position="right",
        plot.background = element_blank(),
        legend.title = element_blank(), 
        axis.text = element_text(size=18, color="black"), 
        axis.title = element_text(size=18, color="black"));pca_all

ggsave(filename="Figures/ITS2/ACR_ITS2_PCA.pdf", plot=pca_acr, dpi=300, width=7, height=5, units="in")
```

This plot isn't very helpful 